proposals/lip-0061.md

```diff
@@ -442,7 +442,7 @@ Every new incoming single commit message `m` is validated as follows:
 
 1. Discard `m` if it is already contained in `nonGossipedCommits` or `gossipedCommits`, i.e., if there is a message `m2` in `nonGossipedCommits` or `gossipedCommits` with `m2.validatorAddress = m.validatorAddress` and `m2.blockID = m.blockID`.
 2. Discard `m` if `m.height <= getMaxRemovalHeight()`.
-3. Discard `m` if `m.height` is not in `{getBFTHeights().maxHeightPrecommitted - COMMIT_RANGE_STORED, ..., getBFTHeights().maxHeightPrecommitted}` and `existBFTParameters(m.height+1)` returns `False` (i.e., `m.height` is not the height of a block authenticating a change of BFT parameters).
+3. Discard `m` if `m.height` is not in `{getBFTHeights().maxHeightPrecommitted - COMMIT_RANGE_STORED, ..., currentHeight}` and `existBFTParameters(m.height+1)` returns `False` (i.e., `m.height` is not the height of a block authenticating a change of BFT parameters). Here `currentHeight` is the height of the latest block added to the blockchain.
```

Comment:
Comment about line 417: `maxHeightFinalized` is used but nowhere defined.

Reply:
Just to make sure I understand: Here we allow validators to sign certificates for non-finalized blocks (in case validator is not lagging behind and the commit satisfies `m.height > getBFTHeights().maxHeightPrecommitted`). 

It the reason that this does not imply that a non-finalized state can be certified, due to the check in `verifyAggregateCommit` (line 588)?

Reply:
Regarding `maxHeightFinalized` I added a reference to its definition after the function.

Regarding the second comment:
> Here we allow validators to sign certificates for non-finalized blocks 

You could phrase it like this, but note that there is no global unique notion of `non-finalized` as this depends on each node's view of the chain. Although we do not punish it, an honest node should never sign single commits for non-finalized blocks.

However, even if malicious nodes were to sign single commits for non-finalized blocks, this is not an attack vector (apart from the spamming of unnecessary data). As you also observe, the function 
`selectAggregateCommit` only considers single commits of height `<= maxHeightPrecommitted` and in `verifyAggregateCommit` we also ensure that this condition is satisfied. Therefore, an aggregate commit included in the blockchain always has to satisfy `aggregateCommit.height <= getBFTHeights().maxHeightPrecommitted`.

Reply:
Ok just wanted to make sure that I follow correctly. It is clear that there is no global unique notion of non-finalized, but indeed the new change allows to sign certificates of non-finalized state (by accident or intentionally..) so wanted to double check that this does not cause any issues.

Can be resolved from my side.